1/ToDoApp/src/backend/resolvers/user/userResolver.js

```diff
@@ -0,0 +1,56 @@
+const { find }= require('lodash');
+const { generateIntID }= require("../../utils.js");
+const { CONFIG }= require("../../config/config");
+const bcrypt = require('bcryptjs')
+const jwt = require('jsonwebtoken')
+
+const users = [{
+  id: 1,
+  name: 'testuser',
+  email: 'your@email.com',
+  password: '$2y$10$3IUx11G0mcJfSpFWn3Lru.xac9OqHDzqLOAhdZovaUyKa2DhgCaOS' // "password"
+},
+{
+  id: 2,
+  name: '2. testuser',
+  email: '2your@email.com',
+  password: '$2y$10$3IUx11G0mcJfSpFWn3Lru.xac9OqHDzqLOAhdZovaUyKa2DhgCaOS' // "password"
+}
+]
+
+const userResolver = {
+  Query: {
+    allUsers: () => users,
+  },
+  Mutation: {
+    signup: (_, {name, email, password}) => {
+      const userID = generateIntID()
+      const userFromEmail = find(users, {email: email});
+      if (userFromEmail) {
+        throw new Error(`This email is already used by another user: ${email}`);
+      }
+      users.push({
+        id: userID,
+        name: name,
+        email: email,
+        password: bcrypt.hash(password, 10),
+      });
+      const token = jwt.sign({id: userID, email: email}, CONFIG.JWT_SECRET, {expiresIn: '1y'});
+      return token;
+    },
+    login: (_, {email, password}) => {
+      const userFromEmail = find(users, {email: email});
+      if (!userFromEmail) {
+        throw new Error(`Couldn’t find a user with the email: ${email}`);
```

Comment:
I think it's best practice to `throw new Error('Incorrect password or email address')` in both cases. That way, an attacker cannot tell if a user is present in the database.